core/src/server/helpers.rs

```diff
@@ -160,6 +160,24 @@ pub fn prepare_error(data: &[u8]) -> (Id<'_>, ErrorCode) {
 	}
 }
 
+/// Figure out if this is a sufficiently complete request that we can extract [`Id`] out of, or just plain
+/// unparseable garbage.
+pub fn prepare_batch_error(data: &[u8]) -> Result<(Vec<Id<'_>>, ErrorCode), ErrorCode> {
+	match serde_json::from_slice::<Vec<&serde_json::value::RawValue>>(data) {
+		Ok(values) => {
+			let ids = values
+				.into_iter()
+				.map(|value| match serde_json::from_str::<InvalidRequest>(value.get()) {
```

Comment:
FYI, because we are using `serde::RawValue` and it doesn't work for untagged enums.

For example

```json
[
        {"jsonrpc": "2.0", "method": "foo", "id": "1"},
        // notification (this shouldn't be answered)
        {"jsonrpc": "2.0", "method": "foo"},
        // invalid call
        {"foo": "boo"},
]

// according the spec this should be returned
[
        {"jsonrpc": "2.0", "result": 7, "id": "1"},
        {"jsonrpc": "2.0", "error": {"code": -32600, "message": "Invalid Request"}, "id": null},
]
```

Strictly speaking we are not following the spec when for example a mixed batch with calls and notifications doesn't work because of the serde::RawValue thingy...

So if we going to change this according to the spec would seems useful to i.e, if a one call fails in batch you would still want the result of the results

With that said, you need check if the `value` is `Request, Notification or InvalidRequest` to do this correctly. If all those fails then regard it as ParseError.

Thus, we would need some test for that as well


Can you add a test for a batch such as:

Reply:
Your code would just return:

```json
[invalid null, invalid null, invalid null]
```

which isn't correct but because of `RawValue` stuff which is tricky for sure ^^